test/unit/http_utils_test.cpp

```diff
@@ -154,6 +154,29 @@ LT_BEGIN_AUTO_TEST(http_utils_suite, standardize_url)
     LT_CHECK_EQ(result, "/abc/pqr");
 LT_END_AUTO_TEST(standardize_url)
 
+LT_BEGIN_AUTO_TEST(http_utils_suite, generate_random_upload_filename)
+    const char* UPLOAD_FILENAME_TEMPLATE = "libhttpserver.XXXXXX";
+
+    #if defined(_WIN32)
+        const char PATH_SEPARATOR = '\\';
+    #else  // _WIN32
+        const char PATH_SEPARATOR = '/';
+    #endif  // _WIN32
+
```

Comment:
Okay, if <filesystem> is used from C++17, then this is not needed anymore.  There's `std::filesystem::path` then which has operators for building paths, so users don't have to worry about / or \ anymore.  See https://en.cppreference.com/w/cpp/filesystem/path for reference.